packages/config-service/src/services/globalConfig.ts

```diff
@@ -742,5 +763,44 @@ export class GlobalConfig {
       required: false,
       defaultValue: null,
     },
-  };
+} satisfies { [key: string]: ConfigProperty };
+
+export type ConfigKey = keyof typeof _CONFIG;
+
+export class ConfigService {
```

Comment:
No need for this extra class. The runtime conversion is already done by `ValidationService.typeCasting`.